src/blocknative.rs

```diff
@@ -0,0 +1,239 @@
+use super::{linear_interpolation, GasPriceEstimating, Transport};
+use crate::CachedResponse;
+use anyhow::{anyhow, Context, Result};
+use serde::Deserialize;
+use std::{
+    convert::TryInto,
+    sync::Arc,
+    time::{Duration, Instant},
+};
+use tokio::{
+    sync::Mutex,
```

Comment:
In this case we can use the std mutex. The reason is that you only need an async mutex if the lock is held while we're also `await`ing an async operation (the lock is held across an await point). Here this doesn't happen so std mutex suffices and is more efficient than an async mutex.

Reply:
Fixed.